src/transformStateWithClones.js

```diff
@@ -8,6 +8,34 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const clonedState = { ...state };
+  const states = [];
+
+  for (let n = 0; n < actions.length; n++) {
+    switch (actions[n].type) {
+      case 'addProperties':
+        Object.assign(clonedState, actions[n].extraData);
+        break;
+
+      case 'removeProperties':
+        for (let i = 0; i < actions[n].keysToRemove.length; i++) {
+          delete clonedState[actions[n].keysToRemove[i]];
+        }
+
+        break;
+
+      case 'clear':
+        const stateKeys = [ ...Object.keys(clonedState) ];
+
+        for (let i = 0; i < stateKeys.length; i++) {
+          delete clonedState[stateKeys[i]];
+        }
```

Comment:
what about `clonedState = {}` to clear it

Reply:
I... genuinely have no idea why I did not think of that, I feel really stupid now

Reply:
It was not that obvious, don't worry